### PR TITLE
[MB-3968] Get lib/pq working again under Pop 5.x

### PR DESCRIPTION
**Pull request by reggieriser**
## Description

This PR contains changes needed to get Pop 5.x and our the `lib/pq` driver working and passing all tests.

## Reviewer Notes

- Note that this PR is requesting a merge into rr-MB-2467-pop5 (the long-running branch for the Pop upgrade), not into master. Eventually we will merge the long-running branch into master after all tests are passing and we deploy/test on experimental.
- I also upgraded Pop to the recently released 5.3.1 which has a preloading bug fix we'll need.
- Previous PRs had changed some error codes to look for the `pgx` versions, but I'm changing them back to the `pq` versions here since we're going to stay with `lib/pq` as an intermediate step to getting on Pop 5.x.
- Because of some changes under the hood for Pop, the Pop setup for the tests had to be modified to explicitly set the SSL mode.
- Because Pop defaults to the `jackc/pgx` driver now, we have to explicitly pass the `lib/pq` driver (i.e., `postgres`) to the Pop setup for the tests.  The rest of the code uses a `custompostgres` driver that already wraps `lib\pq`.

## Setup

- `make server_test` -- ensure all tests are passing 100%
- `make server_run` and `make client_run`, then try making a test move with the app and looking at it from the office side.

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3968) for this change

### CODE REVIEW

**Review by gidjin — COMMENTED** [review]
Changes look good, tests passed locally. I was able to create a move but I can't seem to see it in the office. Maybe we can walk through that together to get the approval on this?

File: pkg/testingsuite/pop_suite.go [hunk 1]
```diff
@@ -127,15 +130,22 @@ func NewPopTestSuite(packageName PackageName) PopTestSuite {
...
+	dbSSLMode := envy.Get("DB_SSL_MODE", "disable")
+
+	dbOptions := map[string]string{
+		"sslmode": dbSSLMode,
+	}
 
 	log.Printf("package %s is attempting to connect to database %s", packageName.String(), dbNameTest)
 	primaryConnDetails := pop.ConnectionDetails{
 		Dialect:  dbDialect,
+		Driver:   "postgres",
```
[the remark this answers is not in the document]
Comment (reggieriser) [reply pkg/testingsuite/pop_suite.go:142]: Both the tests and the app code will continue to use the `lib/pq` driver for this feature.  The default in Pop changed from `lib/pq` to `jackc/pgx`, so we have to now specify the "postgres" (aka, `lib\pq`) driver where we were just taking the default before.  The app code already specified a ["custompostgres" driver](https://github.com/transcom/mymove/search?q=custompostgres) -- a wrapper around `lib/pq` that we appear to do some IAM magic with -- so nothing had to change there.

The transition to `jackc\pgx` moved to a [different feature](https://dp3.atlassian.net/browse/MB-3965) so we can still make progress on upgrading Pop.

**Comment by reggieriser** [comment]
> Changes look good, tests passed locally. I was able to create a move but I can't seem to see it in the office. Maybe we can walk through that together to get the approval on this?

I ran into this too but it appears to be caused by a recent change to restrict what you see in the TOO/TIO UI to what's associated with your transportation office.  I can walk you through what I discovered.

**Review by gidjin — APPROVED** [review]
Server Tests passed, and I was able to create a move and take it all the way through invoicing stage once I updated the transportation office of my TOO and TIO accounts!

**Comment by lynzt** [comment]
LGTM... I'm stuck at the same point John G was (seeing the move in the office app), but since it's working for him, 👍

**Pull request merged**